packages/dds/task-manager/src/taskManager.ts

```diff
@@ -312,25 +317,69 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         return lockAcquireP;
     }
 
-    public abandon(taskId: string) {
-        if (!this.connected) {
-            throw new Error(`Attempted to abandon in disconnected state: ${taskId}`);
+    public subscribeToTask(taskId: string) {
+        if (this.subscribed(taskId)) {
+            return;
         }
 
-        // Nothing to do if we're not at least trying to get the lock.
-        if (!this.queued(taskId)) {
-            return;
+        const submitVolunteerOp = () => {
+            this.submitVolunteerOp(taskId);
+        };
+
+        const disconnectHandler = () => {
+            // Wait to be connected again and then re-submit volunteer op
+            this.connectionWatcher.once("connect", submitVolunteerOp);
+        };
+
+        const checkIfAbandoned = (eventTaskId: string) => {
+            if (eventTaskId !== taskId) {
```

Comment:
Another thought here -- abandoning a subscription is silent (no event or rejected promise).  Should there be some event to reveal the state transition?  Not necessary for this PR though

Reply:
I'm not sure if this will be necessary. Subscribing guarantees you will always try to be in queue unless you manually call `abandon()` (or the task is completed in future iterations). So in theory, you should never need to listen to an "unsubscribe" event since it will be triggered by a manual `abandon()` call.